message.go

```diff
@@ -45,7 +45,15 @@ func (m *Message) encode(pe packetEncoder) error {
 	pe.putInt8(attributes)
 
 	if m.Version >= 1 {
-		pe.putInt64(m.Timestamp.UnixNano() / int64(time.Millisecond))
+		timestamp := int64(-1)
+
+		if m.Timestamp.After(time.Unix(0, -1)) {
```

Comment:
looking at the actual implementation (https://golang.org/src/time/time.go?s=29125:29162#L991), did you mean `time.Unix(0, 0)`?